key_gen.go

```diff
@@ -68,12 +78,20 @@ func KeyForPrefixedStringMap(
 		buf.WriteByte(prefixSplitter)
 	}
 
-	sortedKeysLen := len(stringMap)
-	for i := 0; i < sortedKeysLen; i++ {
-		buf.WriteString(keys[i])
+	for i, k := range keys {
+		buf.WriteString(k)
 		buf.WriteByte(keyNameSplitter)
-		buf.WriteString(stringMap[keys[i]])
-		if i != sortedKeysLen-1 {
+
+		// Find and write the value for this key. Rightmost map takes
+		// precdence.
+		for j := len(maps) - 1; j >= 0; j-- {
```

Comment:
(idea for follow up): if instead of storing `[]string`, if we instead had `[]struct{key string, index int}`, dedupe would keep the highest precedence map and we could avoid a loop here.